rfcs/react-components/convergence/custom-styling.md

```diff
@@ -0,0 +1,214 @@
+# RFC: Custom Styling
+
+Geoff Cox
+Ben Howell
+Mason Tejera
+Brandon Thomas
+
+## Summary
+
+Fluent UI React v9 is built today to enforce adopting the Fluent 2 design styles. Partners creating component
+libraries don't know how provide their customers the ability to deviate from Fluent 2 design like they could
+with v0, v7, or v8.
+
+We need to define a custom styling escape hatch mechanism or pattern for these partners.
+
+## Background
+
+Fluent UI React v9 strongly conforms to the Fluent 2 Design to provide consistency in component behavior, layout,
+and style. Most partners are expected to adopt the Fluent 2 Design and remove custom styling that deviates from the
+design direction for Office.
+
+Some of our partner teams are building component libraries on top of Fluent UI React.
+They have customers who need to diverge from the Fluent 2 design for both the partners components and our components.
+Our partners need a way for their customers to customize the specific style elements of components
+(e.g. button border) without undue effort.
+
+v9 provides several mechanisms to customize components:
+
+- custom themes
+- custom design tokens
+- CSS classes & styles built with Griffel
+- component part replacement using slots
+- component recomposition using hooks
+
+These mechanisms have several limitations:
+
+- design tokens don't provide the same granularity of customization as v0, v7, or v8
+- the number of CSS vars has an upper limit of ~300 before performance degrades
```

Comment:
MADS itself has about 200 slots dedicated to color alone. Leaving 100 for everything else from borders to fonts feels very limiting.